GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/CoinStorePageController.cs

```diff
@@ -12,31 +9,30 @@ public class CoinStorePageController : MonoBehaviour
     private int _coinsToPurchase;
 
     public void FiveCoinsOnClick()
-    {   
-        confirmPanel.SetActive(true);
-        _coinsToPurchase = 5;
-        confirmText.text = "Would you like to purchase 5     Coins with $0.99?";
+    {
+        itemCoinsOnClick(5, 0.99f);
     }
     
     public void TenCoinsOnClick()
     {   
-        confirmPanel.SetActive(true);
-        _coinsToPurchase = 10;
-        confirmText.text = "Would you like to purchase 10  Coins with $1.99?";
+        itemCoinsOnClick(10, 1.99f);
     }
     
     public void TwentyCoinsOnClick()
     {   
-        confirmPanel.SetActive(true);
-        _coinsToPurchase = 20;
-        confirmText.text = "Would you like to purchase 20  Coins with $2.99?";
+        itemCoinsOnClick(20, 2.99f);
     }
     
     public void FiftyCoinsOnClick()
     {   
+        itemCoinsOnClick(50, 4.99f);
+    }
+
+    private void itemCoinsOnClick(int coinAmount, float price)
```

Comment:
name this method buyCoins. Also define a class object for coin, which can contain amount, and selling price, or maybe icon.